### PR TITLE
`rover` 0.5.1 -> 0.11.0, with darwin support

### PR DESCRIPTION
**Pull request by aaronArinder**
###### Description of changes

Bumps `rover` to 0.11.0 and adds darwin support for the macOS homies.

###### Things done
- Built on platform(s)
  - [x] x86_64-linux (see the note above)
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by aaronArinder** [comment]
not sure what's up with that dns failure; for posterity, here's the failure:

```
running 3 tests
test latest_plugins_are_valid_versions ... FAILED
test it_has_nix_installer ... ok
test it_has_windows_installer ... ok
failures:
---- latest_plugins_are_valid_versions stdout ----
thread 'latest_plugins_are_valid_versions' panicked at 'could not send HEAD request to https://github.com/apollographql/federation-rs/releases/download/supergraph@v0.37.1/supergraph-v0.37.1-x86_64-unknown-linux-gnu.tar.gz: error sending request for url (https://github.com/apollographql/federation-rs/releases/download/supergraph@v0.37.1/supergraph-v0.37.1-x86_64-unknown-linux-gnu.tar.gz): error trying to connect: dns error: failed to lookup address information: Temporary failure in name resolution', tests/installers.rs:159:13
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
failures:
    latest_plugins_are_valid_versions
```

there are two other calls being made in that test doing the same thing (checking tarballs) at the same github repo's release page; I'm hoping it truly is just some intermittent dns blip, but going to keep fiddling with it to see if I can figure out what's wrong

**Comment by aaronArinder** [comment]
the `checkFlags` _should_ do the dirt (and it does it when running tests via `cargo`), but the problematic test seems to still run when doing a nix build; I'll look more into it tomorrow

**Comment by aaronArinder** [comment]
rtfm'd myself with `checkFlags` versus `cargoTestFlags` for passing a flag to skip a network-dependent test

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 215180` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>rover</li>
  </ul>
</details>

**Comment by aaronArinder** [comment]
@siraben, I didn't know about that repo! super cool project

**Comment by aaronArinder** [comment]
@ivanbrennan: howdy! hoping this didn't get lost in your inbox, but pinging just in case; I'd like to start helping with rover

**Comment by aaronArinder** [comment]
also, @winterqt, let me know if there's anything funky with the history on this branch (but only if you've the bandwidth--feel free to ignore this message); I want to become a regular contributor/reviewer, but need some help understanding how to reduce noise/friction rather than add to it

File: pkgs/development/tools/rover/default.nix [hunk 1]
```diff
@@ -3,32 +3,47 @@
...
     repo = pname;
     rev = "v${version}";
-    sha256 = "sha256-wBHMND/xpm9o7pkWMUj9lEtEkzy3mX+E4Dt7qDn6auY=";
+    sha256 = "sha256-Ei6EeM0+b3EsMoRo38nHO79onT9Oq/cfbiCZhyDYQrc=";
   };
 
-  cargoSha256 = "sha256-n0R2MdAYGsOsYt4x1N1KdGvBZYTALyhSzCGW29bnFU4=";
+  cargoSha256 = "sha256-+iDU8LPb7P4MNQ8MB5ldbWq4wWRcnbgOmSZ93Z//5O0=";
+
+  buildInputs = [ openssl ] ++ lib.optionals stdenv.isDarwin [
```
Comment (ivanbrennan) [inline pkgs/development/tools/rover/default.nix:24]: Why is openssl needed?

Comment (aaronArinder) [reply pkgs/development/tools/rover/default.nix:24]: shoot, it's not! that was a sloppy copy/paste 😬

Comment (aaronArinder) [reply pkgs/development/tools/rover/default.nix:24]: waiting for a build on another machine to doublecheck x86 builds and then I'll push a commit removing `openssl`

**Review by ivanbrennan — APPROVED** [review]
Thanks!

File: pkgs/development/tools/rover/default.nix [hunk 2]
```diff
@@ -41,10 +55,10 @@ rustPlatform.buildRustPackage rec {
...
 
   meta = with lib; {
-    description = "A CLI for managing and maintaining graphs with Apollo Studio";
+    description = "A CLI for interacting with ApolloGraphQL's developer tooling, including managing self-hosted and GraphOS graphs.";
     homepage = "https://www.apollographql.com/docs/rover";
     license = licenses.mit;
-    maintainers = [ maintainers.ivanbrennan ];
-    platforms = ["x86_64-linux"];
+    maintainers = [ maintainers.ivanbrennan maintainers.aaronarinder ];
+    platforms = ["x86_64-linux" "aarch64-darwin"];
```
Comment (winterqt) [inline pkgs/development/tools/rover/default.nix:62]: This should be `platforms.unix`, or just omit it entirely if it works on both Linux and Darwin. Right now, you're limiting it to just x86_64-linux and aarch64-darwin, when aarch64 and x86_64 versions of those systems.

**Review by winterqt — CHANGES_REQUESTED** [review]
Hi, and welcome!

I have a few things I'd like to note. Overall, these changes look good -- thank you!

1. Please rebase your commits, instead of creating merge commits. It helps us keep a clean history.
2. Can you squash the removal of the OpenSSL dependency so that it's never added in the first place?
3. What happened to 8f0384bb80a4cccdc0c0f32753a96ea408a839b1? Looks like it's gone, but I can't find where it was removed.

There's one or two other things I'd like to touch on, but I'd rather wait until after the commit history is cleaned up.

Thanks again!

**Comment by aaronArinder** [comment]
@stig , @zakame, and @dasJ: sorry for the false alarm, I was doing some branch clean up and messed up the history!

File: pkgs/development/tools/rover/default.nix (hunk 2 above)
Comment (aaronArinder) [reply pkgs/development/tools/rover/default.nix]: done!

**Comment by aaronArinder** [comment]
@winterqt: the history should look sensible and be in a fairly logical order, so hit me up with whatever else you'd like to touch on! and, thanks again for the reviews; I don't know how many folks are committers, but I get the sense that you're carrying a significant share of that work

**Comment by stig** [comment]
Different Stig perhaps? I have no clue what you're referring to I'm afraid 😄

**Comment by aaronArinder** [comment]
whew, yeah, I shouldn't have tried to jam fixing up the branch in before a meeting, it only made it super sloppy--sorry for the wrong ping!

**Pull request merged**